### ISSUE TITLE
Feature Request: Make generated static class partial so it can be easily extended

### ISSUE THREAD

**Issue by VagyokC4** [andrewlock/NetEscapades.EnumGenerators#119]
Any issues with making the generated static class **partial** so that it can be extended with additional extensions methods?

I'd like to do something like the following:

```
[EnumExtensions]
public enum MyEnum
{
    [Group(1)] First,

    [Display(Name = "2nd")]
    [Group(2)] 
    Second,
}
```

```
// Extend the EnumGenerators static class
public static partial class MyEnumExtensions
{
    // Add another extension method to the MyEnumExtensions static class
    public static int GetGroup( this MyEnum code )
    {
        var fieldInfo = code.GetType().GetField(code.ToString());
        var attribute = fieldInfo?.GetCustomAttribute<GroupAttribute>();
  
        if ( attribute != null ) return attribute.Group;
        throw new InvalidOperationException($"Group not defined for {code}");
    }
}
```
assuming the following Group attribute exists
```
[AttributeUsage(AttributeTargets.Field)]
public class GroupAttribute : Attribute
{
    public int Group { get; }

    public GroupAttribute(int group)
    {
        Group = group;
    }
}
```
and be able to use it like this
```
MyEnumExtensions.TryParse("First", out var result);
var grp = MyEnumExtensions.GetGroup(result);
```
making it easy to extend the provided methods that the [EnumExtensions] attribute give us?

**Comment by andrewlock** [comment]
Hey @VagyokC4, the extension methods are _already_ marked partial 😄 e.g. https://github.com/andrewlock/NetEscapades.EnumGenerators/blob/abc942713a615b977223ecaf36ebb9bad138cdeb/tests/NetEscapades.EnumGenerators.Tests/Snapshots/EnumGeneratorTests.CanGenerateEnumExtensionsForFlagsEnum_Params.verified.txt#L23

Is it possible you're trying to extend an extension class that's in a different project? That's not possible in C# unfortunately

**Comment by VagyokC4** [comment]
> Hey @VagyokC4, the extension methods are _already_ marked partial 😄 e.g. https://github.com/andrewlock/NetEscapades.EnumGenerators/blob/abc942713a615b977223ecaf36ebb9bad138cdeb/tests/NetEscapades.EnumGenerators.Tests/Snapshots/EnumGeneratorTests.CanGenerateEnumExtensionsForFlagsEnum_Params.verified.txt#L23
> 
> Is it possible you're trying to extend an extension class that's in a different project? That's not possible in C# unfortunately

Hmm they are on the same project, maybe it is a namespace issue.  Let me try again.

**Comment by andrewlock** [comment]
Did you ever figure this out @VagyokC4? I'm going to close this for now as I think this is already addressed, but feel free to open it again if you're still having issues
